src/test/java/com/project/playlist/service/impl/GetSortedVideosForPlaylistTest.java

```diff
@@ -0,0 +1,115 @@
+package com.project.playlist.service.impl;
+
+import com.project.playlist.exceptions.PlaylistNotFoundException;
+import com.project.playlist.model.Playlist;
+import com.project.playlist.model.PlaylistVideo;
+import com.project.playlist.model.Video;
+import com.project.playlist.repository.PlaylistVideoRepository;
+import com.project.playlist.service.PlaylistService;
+import com.project.playlist.service.PlaylistVideoService;
+import com.project.playlist.service.VideoService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.function.Executable;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+@SpringBootTest
+@ExtendWith(MockitoExtension.class)
+public class GetSortedVideosForPlaylistTest {
+    private PlaylistVideoService playlistVideoService;
+    @Mock
+    private PlaylistService playlistServiceMock;
+    @Mock
+    private VideoService videoServiceMock;
+    @Mock
+    private PlaylistVideoRepository playlistVideoRepositoryMock;
+
+    private Long invalidPlaylistId;
+
+    @BeforeEach
+    void setUp() {
+        playlistVideoService = new PlaylistVideoServiceImpl(videoServiceMock, playlistServiceMock, playlistVideoRepositoryMock);
+    }
+
+    @Test
+    void givenPlaylistWithVideos_whenSortVideosInPlaylist_thenReturnSortedVideosForPlaylistByOrderNumber() {
+        //given
+        final Long correctPlaylistId = 1L;
+        List<Video> actualVideoList;
+        Playlist playlist = new Playlist();
+        playlist.setId(1L);
+        playlist.setName("playlistName");
+        Video video1 = new Video();
+        video1.setId(1L);
+        video1.setName("video1");
+        Video video2 = new Video();
+        video2.setId(1L);
+        video2.setName("video2");
+        Video video3 = new Video();
+        video3.setId(3L);
+        video3.setName("video3");
+        PlaylistVideo playlistVideo1 = new PlaylistVideo(1L, video1, playlist, 1);
+        PlaylistVideo playlistVideo2 = new PlaylistVideo(2L, video2, playlist, 2);
+        PlaylistVideo playlistVideo3 = new PlaylistVideo(3L, video3, playlist, 3);
+        List<PlaylistVideo> playlistVideoList = new ArrayList<>(List.of(playlistVideo1, playlistVideo2, playlistVideo3));
```

Comment:
Here you created sorted list already and in test you check if list is sorted. You can not be sure in that case that your method is working properly.

Create unordered playlist like this:
`List<PlaylistVideo> playlistVideoList = new ArrayList<>(List.of(playlistVideo13, playlistVideo1, playlistVideo2));`

and with this list perform tests